map/Region.dhall

```diff
@@ -0,0 +1,270 @@
+              =
+                {- new-customer flag is there because on aws new customers on
+                -- `ap-northeast-1 & us-west-1` can only access some of the availability zones.
+                -- https://aws.amazon.com/about-aws/global-infrastructure/regions_az/
+                -}
+                λ(_params : Params.Type) →
+                λ(region : Region) →
+                  merge
+                    { af-south-1 = [ "a", "b", "c" ]
+                    , ap-east-1 = [ "a", "b", "c" ]
+                    , ap-northeast-1 =
+                          [ "a", "b", "c" ]
+                        # ( if    _params.new-customer.ap-northeast-1
+                            then  [] : List Text
+                            else  [ "d" ]
+                          )
+                    , ap-northeast-2 = [ "a", "b", "c" ]
+                    , ap-northeast-3 = [ "a", "b", "c" ]
+                    , ap-south-1 = [ "a", "b", "c" ]
+                    , ap-southeast-1 = [ "a", "b", "c" ]
+                    , ap-southeast-2 = [ "a", "b", "c" ]
+                    , ca-central-1 = [ "a", "b", "c" ]
+                    , cn-north-1 = [ "a", "b" ]
+                    , cn-northwest-1 = [ "a", "b", "c" ]
+                    , eu-central-1 = [ "a", "b", "c" ]
+                    , eu-north-1 = [ "a", "b", "c" ]
+                    , eu-south-1 = [ "a", "b", "c" ]
+                    , eu-west-1 = [ "a", "b", "c" ]
+                    , eu-west-2 = [ "a", "b", "c" ]
+                    , eu-west-3 = [ "a", "b", "c" ]
+                    , me-south-1 = [ "a", "b", "c" ]
+                    , sa-east-1 = [ "a", "b", "c" ]
+                    , us-gov-east-1 = [ "a", "b", "c" ]
+                    , us-gov-west-1 = [ "a", "b", "c" ]
+                    , us-east-1 = [ "a", "b", "c", "d", "e", "f" ]
+                    , us-east-2 = [ "a", "b", "c" ]
+                    , us-west-1 =
+                          [ "a", "b" ]
+                        # ( if    _params.new-customer.us-west-1
+                            then  [] : List Text
+                            else  [ "d" ]
+                          )
+                    , us-west-2 = [ "a", "b", "c" ]
+                    }
+                    region
+
+          let render
+              : Params.Type → Region → List Text
+              =
+                {- new-customer flag is there because on aws new customers can
+                -- only access some of the availability zones.
+                -}
+                λ(_params : Params.Type) →
+                λ(region : Region) →
+                  let generate
+                      : Region → List Text
+                      = λ(region : Region) →
+                          Prelude.List.map
+                            Text
+                            Text
+                            (λ(letter : Text) → name region ++ letter)
+                            (letters Params::{=} region)
```

Comment:
please write an assertion for this in at least one case  - i.e. write `let test : assert generate Region.eu-west-1 === ["eu-west-1a", "eu-west-1b", "eu-west-1c" ]` to show that this works as you expect